### DISCUSSION TITLE
issue: Project not building anymore due to "Can't import the named export 'X'" error

### DISCUSSION THREAD

**Discussion by rnnyrk** [react-hook-form/react-hook-form#7461] [Q&A] [answered]
### Version Number

7.22.5

### Codesandbox/Expo snack

https://github.com/rnnyrk/rhf-import-bug

### Steps to reproduce

1. Download the linked repository
2. Run `npm install`
3. Go to `localhost:3000`
4. See the error

### Expected behaviour

I recently tried to update my project to the latest React Hook Form version. Although, since the latest update I run into the errors shown in the log output. Before I'd use version `7.17.1`, which still worked without any problems. 

I'm not sure how to fix this issue. Using a default export doesn't seem like to solve the problem. Also I couldn't find any relevant commits in the change log regarding this, so it seems like I'm missing something migrating to the latest version, or is this a bug?

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
Compiled with problems:X

WARNING in ./src/components/Form.tsx 89:107-117

Can't import the named export 'Controller' (imported as 'Controller') from default-exporting module (only default export is available)


WARNING in ./src/components/Form.tsx 111:41-51

Can't import the named export 'Controller' (imported as 'Controller') from default-exporting module (only default export is available)


WARNING in ./src/components/Form.tsx 135:41-51

Can't import the named export 'Controller' (imported as 'Controller') from default-exporting module (only default export is available)


WARNING in ./src/components/common/form/FormContainer/index.tsx 8:16-23

Can't import the named export 'useForm' (imported as 'useForm') from default-exporting module (only default export is available)


WARNING in ./src/components/common/form/FormContainer/index.tsx 14:38-50

Can't import the named export 'FormProvider' (imported as 'FormProvider') from default-exporting module (only default export is available)
```
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
> "start": "NODE_ENV=development TS_NODE_PROJECT=config/tsconfig.json webpack serve --config config/webpack.config.dev.ts",


hey, @rnnyrk I think this may due to your build config. 

apologize in front 🙏  @evocateur (by pinging you), Would you have any clue why is this happening?

**Comment by rnnyrk** [comment]
Hmm, if it has something to do with the build config, it's interesting that React Hook Form is the only package affected by it. Will try some stuff with the build config, but also really curious to what has changed between 7.17 and 7.21 that is causing this issue.

**Comment by evocateur** [comment]
Definitely smells like a Webpack issue, as RHF itself doesn't _have_ a default export. Perhaps `TsconfigPathsPlugin` is breaking it somehow? Or the babel-loader isn't preserving the right module output?

**Reply by bluebill1049** [reply]
thanks for the help! 🙏

**Comment by rnnyrk** [comment]
Thanks for your reply, @evocateur. I've tried using Webpack' aliases instead `TsconfigPathsPlugin`, unfortunately the same issues occurs. Also `babel-loader` has an exclude for node_modules in the Webpack config, so that wouldn't be my guess either. Or something I'm misunderstanding?

**Comment by rnnyrk** [comment, marked as answer]
@bluebill1049 @evocateur Thanks for thinking along. The solution is in this PR (found by @sandervspl) https://github.com/rnnyrk/rhf-import-bug/pull/1/files

Might delete the repository later, so if anyone runs into the issue, try:
```
      {
        test: /\.m?t?j?sx?$/,
        exclude: /node_modules/,
        use: 'babel-loader',
      },
```

**Reply by lonalee** [reply]
The codes should be in webpack.config, module.rules? @rnnyrk

**Reply by rnnyrk** [reply]
@lonalee Yes it should be in your Webpack config ;)
